anx/provider/utils/utils.go

```diff
@@ -22,21 +22,23 @@ func IsNotFoundError(err error) bool {
 	return false
 }
 
-// PanicIfUnauthorized panics if the provided error is due to `http.StatusUnauthorized`.
-// Use this helper to slow down Engine requests by triggering a CrashLoopBackoff.
+// IsUnauthorizedOrForbiddenError returns true for Unauthorized or Forbidden http errors
+// otherwise it returns false
 // NOTE: This helper only works for errors returned by go-anxcloud.
-func PanicIfUnauthorized(err error) {
+func IsUnauthorizedOrForbiddenError(err error) bool {
 	if err == nil {
-		return
+		return false
 	}
 
 	var (
 		genericAPIClientError api.HTTPError
 		legacyAPIClientError  *client.ResponseError
 	)
 
-	if (errors.As(err, &genericAPIClientError) && genericAPIClientError.StatusCode() == http.StatusUnauthorized) ||
-		(errors.As(err, &legacyAPIClientError) && legacyAPIClientError.Response.StatusCode == http.StatusUnauthorized) {
-		panic("Engine responded with http.StatusUnauthorized. Please check the configured API token.")
-	}
+	return (errors.As(err, &genericAPIClientError) && (genericAPIClientError.StatusCode() == http.StatusForbidden || genericAPIClientError.StatusCode() == http.StatusUnauthorized)) ||
+		(errors.As(err, &legacyAPIClientError) && (legacyAPIClientError.Response.StatusCode == http.StatusForbidden || legacyAPIClientError.Response.StatusCode == http.StatusUnauthorized))
+
 }
+
+// ErrUnauthorizedForbiddenBlocked is returned if an operation is blocked due to a recent unauthorized or forbidden request
+var ErrUnauthorizedForbiddenBlocked = errors.New("operation currently blocked due to client side unauthorized/forbidden request limiting")
```

Comment:
maybe call this `ErrUnauthorizedForbiddenBackoff`, because that's what we use it for?